### PR TITLE
Added Option to provide selector for MouseAction

### PR DESCRIPTION
**Pull request by jayandran-Sampath**
Added option to provide Selector with mouseAction.

Related #893

### CODE REVIEW

File: lib/taiko.js [hunk 1]
```diff
@@ -1660,8 +1660,41 @@ async function highlight(selector, ...args) {
...
+      left,
+      right,
+    } = await domHandler.getBoundingClientRect(elem);
+    coordinates.x = coordinates.x + parseInt(left);
+    coordinates.y = coordinates.y + parseInt(top);
+  } else if (actions.includes(selector)) {
+    coordinates = action;
+    action = selector;
+  } else {
+    throw Error('Please provide a valid parameter');
```
Comment (sriv) [inline lib/taiko.js]: the error message can perhaps be a little more specific. for instance in this case there are the following checks:

```
isSelector(selector) ||
      isString(selector) ||
      Number.isInteger(selector)) &&
    !actions.includes(selector)
```

if this was refactored to have an inverted `if` then the error message can pass specific details. ex.

```
if (!isSelector(selector)) throw Error('not a valid selector: ' + selector);
if (!isString(selector)) throw Error('expected selector to be of type 'String': ' + selector)
...
etc
...
```

Comment (jayandran-Sampath) [reply lib/taiko.js]: sure @sriv i will do that and commit it

File: lib/taiko.js [hunk 2]
```diff
@@ -1669,11 +1672,16 @@ async function mouseAction(
...
-    (isSelector(selector) ||
-      isString(selector) ||
-      Number.isInteger(selector)) &&
-    !actions.includes(selector)
+    !actions.includes(selector) &&
+    !isSelector(selector) &&
+    !isString(selector)
   ) {
+    throw Error(
+      'Kindly pass valid selector/string, Invalid Selector value passed : ' +
```
Comment (sriv) [inline lib/taiko.js]: please remove "Kindly pass valid selector/string," 

The first few characters of error message are most important, and it helps to keep them crisp.

**Comment by sriv** [comment]
@jayandran-Sampath the tests are now failing for valid reasons:

https://github.com/getgauge/taiko/pull/951/checks?check_run_id=346234131

```
ReferenceError: element is not defined
            at mouseAction (/home/runner/work/taiko/taiko/lib/taiko.js:1614:18)
            at module.exports.<computed> (/home/runner/work/taiko/taiko/lib/taiko.js:3511:36)
            at Object.<anonymous> (tests/click.js:76:11)
```

I believe this is coming from https://github.com/getgauge/taiko/pull/951/files#diff-c626d926eeff4a14b1c1f5114bcc21c5R1614

**Pull request merged**